packages/api/src/utils/stripe.js

```diff
@@ -0,0 +1,378 @@
+import Stripe from 'stripe'
+import { randomString } from './billing.js'
+
+/**
+ * @typedef {import('stripe').Stripe} StripeInterface
+ */
+
+/**
+ * @typedef {import("./billing-types").BillingService} BillingService
+ */
+
+/**
+ * @typedef {object} StripeComForBillingService
+ * @property {Pick<StripeInterface['paymentMethods'], 'attach'>} paymentMethods
+ * @property {Pick<StripeInterface['setupIntents'], 'create'>} setupIntents
+ * @property {Pick<StripeInterface['customers'], 'retrieve'|'update'>} customers
+ */
+
+/**
+ * A BillingService that uses stripe.com
+ */
+export class StripeBillingService {
+  /**
+   * @param {StripeComForBillingService} stripe
+   */
+  static create (stripe) {
+    return new StripeBillingService(stripe)
+  }
+
+  /**
+   * @protected
+   * @param {StripeComForBillingService} stripe
+   */
+  constructor (stripe) {
+    /** @type {StripeComForBillingService}  */
+    this.stripe = stripe
+    /**
+     * @type {BillingService}
+     */
+    const instance = this // eslint-disable-line
```

Comment:
do we need this?

Reply:
https://github.com/microsoft/TypeScript/issues/30156#issuecomment-952667829

Reply:
I made it nicer, one line, no eslint-disable 67deec440f4849cf36583e0c995e2bd31d882903